Source/FORMDataSource.m

```diff
@@ -394,6 +394,24 @@ - (BOOL)isEnabled {
     return !_disabled;
 }
 
+- (void)collapseAllGroupsForCollectionView:(UICollectionView *)collectionView {
+    NSMutableArray *indexPaths = [NSMutableArray new];
+    [self.collapsedGroups removeAllObjects];
+    
+    int groupCount = 0;
+    for (FORMGroup *formGroup in self.formData.groups) {
```

Comment:
could you use 

`[self.formData.groups enumerate....]` ? It would give you the index instead of doing `groupCount++`